### PR TITLE
feat: Add IPAddress class (independent of dart:io)

### PR DESCRIPTION
**Pull request by nielsenko**
## Description

This PR adds IP address handling to Relic, providing support for IPv4 and IPv6 addresses with CIDR notation.

Core Types:
- `IPAddress` - Sealed base class for all IP addresses
- `IPv4Address` - IPv4 implementation with dotted-decimal notation
- `IPv6Address` - IPv6 implementation with hexadecimal notation and compression

Factory Methods:
- `IPAddress.parse` - Automatic type detection, supports CIDR notation
- `IPAddress.fromBytes` - Create from raw network-order bytes and prefix length
- `IPv4Address.fromOctets` - Create IPv4 from octets
- `IPv4Address.fromInt` - Create IPv4 from 32-bit integer
- `IPv6Address.fromHextets` - Create IPv6 from 8 hextets
- `IPv6Address.fromSegments` - Create IPv6 from 16-bit segments

CIDR Operations:
- `network` - Get network address for a subnet
- `broadcast` - Get broadcast address
- `contains(IPAddress)` - Check if an IP is within a subnet
- `withPrefixLength(int)` - Create new IP with different prefix

Implementation:
- Immutable design with network byte order (big-endian)
- Lazy-computed and cached properties
- Input validation with error messages
- Endian-aware byte conversion extensions

Why This Change is Necessary:
This allows us to handle IP addresses in Relic independent from dart:io.

IP address handling enables:
- Parsing client addresses from headers (X-Forwarded-For, Forwarded, etc.)
- IP-based access control and rate limiting
- Subnet-based routing and filtering
- Logging and analytics

## Related Issues

- Closes: #? 

## Pre-Launch Checklist

- [x] This update focuses on a single feature or bug fix. (For multiple fixes, please submit separate PRs.)
- [x] I have read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code using [dart format](https://dart.dev/tools/dart-format).
- [ ] I have referenced at least one issue this PR fixes or is related to.
- [x] I have updated/added relevant documentation (doc comments with `///`), ensuring consistency with existing project documentation.
- [x] I have added new tests to verify the changes.
- [ ] All existing and new tests pass successfully.
- [ ] I have documented any breaking changes below.

## Breaking Changes

- [x] No breaking changes.

This is a new feature addition with no impact on existing functionality.

## Additional Notes

Test Coverage:
- IPv4 and IPv6 parsing, validation, and edge cases
- CIDR notation parsing and validation
- Subnet operations (network, broadcast, contains)
- Factory methods and static instances
- Equality, hashCode, and immutability
- Byte conversion extensions

Standards: RFC 791 (IPv4), RFC 4291 (IPv6), RFC 5952 (IPv6 text), RFC 4632 (CIDR)

Known issues:
- Currently doesn't support IPv4 embedding in IPv6 (see #97)
- Missing `subnetMask` (per design/choice, but trivial to add).
- Missing `hostRange` (iterator, or range)

### CODE REVIEW

File: lib/src/adapter/ip_address.dart [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+    final int a,
+    final int b,
+    final int c,
+    final int d, {
+    final int? prefixLength,
+  }) {
+    final bytes = <int>[a, b, c, d];
+    for (final segment in bytes) {
+      if (segment < 0 || segment > 0xFF) {
+        throw ArgumentError('Invalid IPv4 segment: $segment');
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply lib/src/adapter/ip_address.dart]: Fixed in [448b9f9](https://github.com/serverpod/relic/pull/272/commits/448b9f949c3368238249d11288e1458a4fdf2143)

**Comment by nielsenko** [comment]
> We also have a hard dependency on io for the static file things, this will prevent us from running in WASM right? Is there any reason for adding an override for this that will make us incompatible with other solutions?
> 
> Perhaps we should have an adapter for the static things as well and move that into our io adapter?

I agree. `StaticHandler` is inherently dependent on dart:io. I would like to make relic into a monorepo of multiple packages - one being `relic_io`.

I would like to keep that separate from this PR though.

File: lib/src/adapter/ip_address.dart [hunk 2]
```diff
@@ -0,0 +1,563 @@
+import 'dart:typed_data';
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply lib/src/adapter/ip_address.dart]: This PR only adds the IPAddress feature. A follow-up PR will use it on the adapter interface, via an abstract`ConnectionInfo` class, that will be implemented by the various adapters (ie. `IOAdapter` for now).

[the remark this answers is not in the document]
Comment (SandPod) [reply lib/src/adapter/ip_address.dart]: Wrt. mono-repo - Won't this be a core part of Relic that is then adapted for different underlying solutions? Therefore, it should probably be included in the core library? Or am I missing something?

**Comment by nielsenko** [comment]
> > We also have a hard dependency on io for the static file things, this will prevent us from running in WASM right? Is there any reason for adding an override for this that will make us incompatible with other solutions?
> > Perhaps we should have an adapter for the static things as well and move that into our io adapter?
> 
> I agree. `StaticHandler` is inherently dependent on dart:io. I would like to make relic into a monorepo of multiple packages - one being `relic_io` (since lower sdk bound is now ^3.7.0 we have workspace support).
> 
> I would like to keep that separate from this PR though.

@coderabbitai Could you create a new issue to reorganize code into a multi-package mono-repo?
